components/server/src/prebuilds/bitbucket-server-app.ts

```diff
@@ -105,68 +107,69 @@ export class BitbucketServerApp {
         }
     }
 
-    protected async handlePushHook(
+    private async handlePushHook(
         ctx: TraceContext,
         user: User,
         payload: PushEventPayload,
         event: WebhookEvent,
-    ): Promise<StartPrebuildResult | undefined> {
+    ): Promise<void> {
         const span = TraceContext.startSpan("Bitbucket.handlePushHook", ctx);
         try {
             const cloneURL = this.getCloneUrl(payload);
-            const { user: projectOwner, project } = await this.findProjectAndOwner(cloneURL, user);
-            if (!project) {
-                throw new ApplicationError(
-                    ErrorCodes.NOT_FOUND,
-                    `Project not found. Please add '${cloneURL}' as a project.`,
-                );
-            }
+            const projects = await this.projectService.findProjectsByCloneUrl(user.id, cloneURL);
+            for (const project of projects) {
```

Comment:
I think it would be good to guard per project as well, otherwise if anything goes south with the first project at hand, e.g. context parsing, the other projects won't be processed. 
(same applies to the other *-app.ts)